Spatial.gd

```diff
@@ -319,6 +328,26 @@ func decode_context_packet(spb: StreamPeerBuffer):
 	# vs radians. 70deg = 1.22173rad and 25deg = 0.4363323rad. We should redo
 	# this to just be a radian to degree conversion.
 
+	# Calculations to dynamically place the main icon/button
+	# The left- and right-margin can be calculated from the ui-scale combined with the preset icon width and the desired position.
+	self.ui_size = int(identity["uisz"])
+	# If the value is not part of the dictionary use the "normal" size.
+	if !self.icon_size_preset.has(self.ui_size):
+		self.ui_size = 1
+
+	var button_position_max = floor(OS.window_size.x / self.icon_size_preset[self.ui_size]) - 1
+	var button_margin_left = $Control/GlobalMenuButton.margin_left
+	# Make sure the expected position is inside the window.
+	if (self.button_position > button_position_max):
+		button_margin_left = self.icon_size_preset[self.ui_size] * button_position_max
+	else:
+		button_margin_left = self.icon_size_preset[self.ui_size] * self.button_position
+
+	$Control/GlobalMenuButton.margin_left = button_margin_left
+	$Control/GlobalMenuButton.margin_right = button_margin_left + self.icon_size_preset[self.ui_size]
```

Comment:
Setting the right margin might not be necessary, the right margin might be being ignored here due to the anchors. Not really important but if it is not necessary might as well remove it.

Reply:
Not setting the right margin caused the button to disappear on scale `large` and `largest` .
I will check this again.

Reply:
The function `set_minimal_mouse_block()` will not work if the left margin is greater than the right margin.
So we have to set both or rework said function.

Reply:
Makes sense. Ignore this comment then.